### ISSUE TITLE
The Crashlytics build ID is missing.

### ISSUE THREAD

**Issue by liuzhenLz** [xamarin/GooglePlayServicesComponents#642]
The Crashlytics build ID is missing. This occurs when Crashlytics tooling is absent from your app's build configuration. Please review Crashlytics onboarding instructions and ensure you have a valid Crashlytics account.

**Comment by dtaylorus** [comment]
I'm getting this same exception on startup after installing the package Xamarin.Firebase.Crashlytics in a dotnet 6 project.

````
Java.Lang.RuntimeException
  Message=Unable to get provider com.google.firebase.provider.FirebaseInitProvider: java.lang.IllegalStateException: The Crashlytics build ID is missing. This occurs when Crashlytics tooling is absent from your app's build configuration. Please review Crashlytics onboarding instructions and ensure you have a valid Crashlytics account.
````

I think this means the `google-services.json` file is not being processed. I have added the file to my project and set the build action to GoogleServicesJson:

    <GoogleServicesJson Include="google-services.json" />

Is there any other step I'm missing?

**Comment by dtaylorus** [comment]
Yes, there was another missing step. I added the following to my Strings.xml resources file to resolve the issue:

````
<resources>
  <string name="com.crashlytics.android.build_id">1</string>
</resources>
````

**Comment by tranb3r** [comment]
I've solved it by adding
`<string name="com.google.firebase.crashlytics.mapping_file_id">none</string>`

**Comment by Desintegrator** [comment]
What does crashlytics build_id mean?

**Comment by oscore-john** [comment]
> Yes, there was another missing step. I added the following to my Strings.xml resources file to resolve the issue:
> 
> ```
> <resources>
>   <string name="com.crashlytics.android.build_id">1</string>
> </resources>
> ```

this worked for me too, although I also would like to know what the build_id is "supposed" to be.

**Comment by DavidJinius** [comment]
Adding the string resource solved my problem. I don't know why it works, but clearly, this step should be added to Get Started with Firebase Crashlytics on the FIrebase website.

**Comment by waheez** [comment]
yes adding this line works, also a AI response to this workaround, and I agree to it:
I apologize for the confusion, and I'm glad you were able to resolve the issue by adding the mentioned line. The line <string name="com.crashlytics.android.build_id">1</string> appears to be a workaround to address the missing Crashlytics build ID error in Xamarin.Android projects.

Adding that line manually provides a dummy build ID value, allowing the Crashlytics initialization to proceed without encountering the specific error. While this workaround might resolve the issue in some cases, it's important to note that it deviates from the standard Crashlytics integration process.

It's recommended to use the official integration steps provided by Firebase and the Xamarin.Firebase.Crashlytics library. However, if you continue to experience difficulties with the standard integration, you can consider using the workaround as a temporary solution.

Please keep in mind that workarounds like this may not provide the full functionality or stability of the official integration. It's advisable to monitor the official documentation, forums, and updates from Firebase and Xamarin to ensure you are using the most up-to-date and reliable integration approach.

If you have any further questions or need additional assistance, please let me know.

**Comment by headintheclouds21** [comment]
I'm experiencing this issue in my NET MAUI project. I saw @oscore-john had fixed the project by adding that line to "Strings.xml". I don't have a "Strings.xml" file in my project so I created one under Platforms -> Android -> Resources -> values -> *Added Strings.xml here* and I'm still receiving the same error. Am I doing something wrong? It seems like others got this working. Any help would be greatly appreciated.

**Comment by the-crow-club** [comment]
> I'm experiencing this issue in my NET MAUI project. I saw @oscore-john had fixed the project by adding that line to "Strings.xml". I don't have a "Strings.xml" file in my project so I created one under Platforms -> Android -> Resources -> values -> _Added Strings.xml here_ and I'm still receiving the same error. Am I doing something wrong? It seems like others got this working. Any help would be greatly appreciated.

I did same as you im not getting this error any more... make sure the build action is  set to none for Strings.xml

**Comment by headintheclouds21** [comment]
@the-crow-club Thanks for the advice! I was also able to get it working

**Comment by neubami94** [comment]
After several hours i finally figured it out. I used ./gradlew app:dependencies command to list the dependency tree just to notice, that one out of date library used older version of crashlytics library. I updated it and the issue was finally gone!

**Comment by moljac** [comment]
Since there is workaround I am closing this issue.

**Comment by bimix** [comment]
> Yes, there was another missing step. I added the following to my Strings.xml resources file to resolve the issue:
> 
> ```
> <resources>
>   <string name="com.crashlytics.android.build_id">1</string>
> </resources>
> ```

cant belive this is also required in Expo.
